Add Vola Tab Report Binder script

szero opened the pull request:
When you have report tab bound to a given room and lain decides to
nuke your connection, the report tab will be bound to a random room
you have currently open, that regains connection first. This script
will bind the report room to one of your choosing by typing `/bindreports`
command in chat. The report room will be set even after browser restart.

szero committed:
add vola tab report binder script

RealDolos commented on volatabreportbinder.user.js:
```diff
@@ -0,0 +1,44 @@
+  console.log("running", GM.info.script.name, GM.info.script.version, dry.version);
+  const ke = new KeyboardEvent("keydown", {
+    key: "Enter", keyCode: "13"
+  });
```
I cri

RealDolos commented on volatabreportbinder.user.js:
```diff
@@ -0,0 +1,44 @@
+  const ke = new KeyboardEvent("keydown", {
+    key: "Enter", keyCode: "13"
+  });
+  const bindtab = "bindmehere";
```
I cri moar

RealDolos commented on volatabreportbinder.user.js:
```diff
@@ -0,0 +1,44 @@
+    key: "Enter", keyCode: "13"
+  });
+  const bindtab = "bindmehere";
+  const ci = document.getElementById("chat_input");
```
AND MOAR!

RealDolos commented on volatabreportbinder.user.js:
```diff
@@ -0,0 +1,44 @@
+  }();
+  new class extends dry.MessageFilter {
+    showMessage(orig, nick, message, options) {
+      let tabbind = localStorage.getItem(bindtab);
```
calling localstorage.getItem a lot of times, like a pleb... tsk

RealDolos commented on volatabreportbinder.user.js:
```diff
@@ -0,0 +1,44 @@
+  const ci = document.getElementById("chat_input");
+  new class extends dry.Commands {
+    bindreports() {
+      localStorage.setItem(bindtab, window.location.href);
```
why not use the actual roomid?

RealDolos commented on volatabreportbinder.user.js:
```diff
@@ -0,0 +1,44 @@
+    showMessage(orig, nick, message, options) {
+      let tabbind = localStorage.getItem(bindtab);
+      if ("Log" === nick && tabbind === window.location.href) {
+        if (message[0].hasOwnProperty("value")) {
```
two ifs---nested, tsk

RealDolos replied:
not returning early... tsk

RealDolos commented on volatabreportbinder.user.js:
```diff
@@ -0,0 +1,44 @@
+      let tabbind = localStorage.getItem(bindtab);
+      if ("Log" === nick && tabbind === window.location.href) {
+        if (message[0].hasOwnProperty("value")) {
+          let match = message[0].value.match(`${dry.exts.user.info.nick} joined`);
```
uh, uwutm8

RealDolos commented on volatabreportbinder.user.js:
```diff
@@ -0,0 +1,44 @@
+        if (message[0].hasOwnProperty("value")) {
+          let match = message[0].value.match(`${dry.exts.user.info.nick} joined`);
+          if (!!match && !!dry.exts.user.info.staff) {
+            ci.value = "/reports";
```
see above

RealDolos commented on volatabreportbinder.user.js:
```diff
@@ -0,0 +1,44 @@
+      if ("Log" === nick && tabbind === window.location.href) {
+        if (message[0].hasOwnProperty("value")) {
+          let match = message[0].value.match(`${dry.exts.user.info.nick} joined`);
+          if (!!match && !!dry.exts.user.info.staff) {
```
.staff? not .admin? why at all?!

RealDolos commented on volatabreportbinder.user.js:
```diff
@@ -0,0 +1,44 @@
+  new class extends dry.MessageFilter {
+    showMessage(orig, nick, message, options) {
+      let tabbind = localStorage.getItem(bindtab);
+      if ("Log" === nick && tabbind === window.location.href) {
```
ROOMIDDDDDDD

RealDolos commented on volatabreportbinder.user.js:
```diff
@@ -0,0 +1,44 @@
+    bindreports() {
+      localStorage.setItem(bindtab, window.location.href);
+      dry.appendMessage("Binder", "Reports are bound to this tab now");
+      ci.value = "/reports";
```
herro, let's duplicate codez!

RealDolos commented on volatabreportbinder.user.js:
```diff
@@ -0,0 +1,44 @@
+  new class extends dry.Commands {
+    bindreports() {
+      localStorage.setItem(bindtab, window.location.href);
+      dry.appendMessage("Binder", "Reports are bound to this tab now");
```
no, they are not. They are bound to this location

RealDolos commented on volatabreportbinder.user.js:
```diff
@@ -0,0 +1,44 @@
+      }
+    }
+  }();
+});
```
>missing newline

RealDolos commented on volatabreportbinder.user.js:
```diff
@@ -0,0 +1,44 @@
+            ci.value = "/reports";
+            ci.dispatchEvent(ke);
+          }
+        }
```
herro, imma bracket pyramid!

RealDolos reviewed (changes requested):
file name: admin.reportsbinder.user.js or something plox

szero committed:
address review issues

szero commented:
Review changes did't auto-address themselves on push because I changed the filename, lul.

szero committed:
simplify binding logic

szero committed:
even better binding logic and some globals

szero committed:
make localStorage.getItem queryable

now the tab will actually change if we had reports bound to a specific room and want them bound to another
